Q1. Write a Java program to take two numbers as input and divide them. Handle 
ArithmeticException (divide by zero). 

import java.util.Scanner;

public class DivideNumbers {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter first number: ");
        int num1 = sc.nextInt();

        System.out.print("Enter second number: ");
        int num2 = sc.nextInt();

        try {
            int result = num1 / num2;
            System.out.println("Result = " + result);
        } catch (ArithmeticException e) {
            System.out.println("Error: Division by zero is not allowed!");
        }

        sc.close();
    }
}

 Q2. Write a program to handle ArrayIndexOutOfBoundsException when accessing an array 
element. 

import java.util.Scanner;

public class DivideNumbers {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter first number: ");
        int num1 = sc.nextInt();

        System.out.print("Enter second number: ");
        int num2 = sc.nextInt();

        try {
            int result = num1 / num2;
            System.out.println("Result = " + result);
        } catch (ArithmeticException e) {
            System.out.println("Error: Division by zero is not allowed!");
        }

        sc.close();
    }
}

Q3. Create a program to accept age from the user. Throw and catch a user-defined exception if age is less than 18.

import java.util.Scanner;

// Custom exception class
class AgeException extends Exception {
    public AgeException(String message) {
        super(message);
    }
}

public class AgeCheck {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter your age: ");
        int age = sc.nextInt();

        try {
            if (age < 18) {
                throw new AgeException("Age must be 18 or above!");
            } else {
                System.out.println("Valid age: You are eligible.");
            }
        } catch (AgeException e) {
            System.out.println("Exception Caught: " + e.getMessage());
        }

        sc.close();
    }
}
